### DISCUSSION TITLE
Access token generated by Management API has too long lifetime

### DISCUSSION THREAD

**Discussion by char000** [Azure/api-management-developer-portal#2585] [Feature request]
**Every bug report should have precise description and reproduction steps; console traces or source code references are appreciated.**

**For assistance requests, contact Azure support or submit a post [on Stack Overflow](http://aka.ms/apimso). We don't provide support through GitHub Issues. Feature requests can be raised [on the Azure Feedback Forum](https://aka.ms/apimwish).**

## Description

We are using signin delegation for signing user into developer portal.

1. After user is authenticated by our backend, our backend generates SAS token with lifetime of 3 minutes and then redirect the user to developer portal (/signin-sso) with that SAS token.

2. Developer portal uses SAS token to call management API endpoints to get a list of products/apis, etc

3. Management API endpoints (specifically, [this API](https://docs.microsoft.com/en-us/rest/api/apimanagement/2021-01-01-preview/user-identities/list)) returns json response along with a new token in **Ocp-Apim-Sas-Token** header with lifetime of 1hr
 
We would like to have expiry of token generated in step 3 to be 15-20min.

Support request ID:  2109290060001844

**Comment by ghost** [comment]
@char000, thank you for opening this issue. We will triage it within the next few business days.

**Comment by azaslonov** [comment]
Hi @char000, right this is not possible to configure the token lifetime currently. I'm leaving it as a feature request.

**Comment by ghost** [comment]
@char000, thank you for requesting this feature.

**Comment by char000** [comment]
@mikebudzynski  Is there any update about this request?

**Comment by KIT-GregC** [comment]
> @mikebudzynski Is there any update about this request?

I would like to see this feature too.
